tests/fixtures/mod.rs

```diff
@@ -76,3 +81,85 @@ pub fn tmpdir() -> TempDir {
 pub fn port() -> u16 {
     free_local_port().expect("Couldn't find a free local port")
 }
+
+/// Run miniserve as a server; Start with a temporary directory, a free port and some
+/// optional arguments then wait for a while for the server setup to complete.
+#[fixture(args=&[] as &[&str])]
+#[allow(dead_code)]
+pub fn server<S>(args: impl IntoIterator<Item = S>) -> TestServer
+where
+    S: AsRef<std::ffi::OsStr>,
+{
+    let port = port();
+    let tmpdir = tmpdir();
+    let child = Command::cargo_bin("miniserve")
+        .expect("Couldn't find test binary")
+        .arg(tmpdir.path())
+        .arg("-p")
+        .arg(port.to_string())
+        .args(args)
+        .stdout(Stdio::null())
+        .spawn()
+        .expect("Couldn't run test binary");
+
+    sleep(Duration::from_secs(1));
```

Comment:
How about we switch this fairly annoying manual wait to something that's faster and less error prone in this PR as well?

Something like:
```
        // Wait a max of 1s for the port to become available.
        let start_wait = Instant::now();
        while start_wait.elapsed().as_secs() < 1
            && !port_check::is_port_reachable(format!("localhost:{}", port))
        {
            sleep(Duration::from_millis(100));
        }
```